rpc/client.go

```diff
@@ -215,6 +259,14 @@ func (c *Client) close() {
 // or unhealthy and reconnecting appropriately until either the Client or the
 // supplied channel is closed.
 func (c *Client) runHeartbeat(retryOpts retry.Options) {
+	healthReceived := c.healthReceived
+	setHealthReceived := func() {
+		if healthReceived != nil {
+			close(healthReceived)
+			healthReceived = nil
```

Comment:
Race in what way? If someone is blocked on `c.healthReceived` they will be woken by the close. Note that `c.healthReceived` is never set to nil.